File: scanit_app/lib/models/grocery_item.dart
class GroceryItem {
  final String id;
  final String name;
  final int quantity;
  final String unit;

  GroceryItem({
    required this.id,
    required this.name,
    required this.quantity,
    required this.unit,
  });

  factory GroceryItem.fromJson(Map<String, dynamic> json) {
    return GroceryItem(
      id: json['_id'],
      name: json['name'],
      quantity: json['quantity'],
      unit: json['unit'],
    );
  }
}

File: scanit-backend/models/GroceryItem.js
const mongoose = require('mongoose');

const GroceryItemSchema = new mongoose.Schema({
  userId: { type: String, required: true },
  name: { type: String, required: true },
  quantity: { type: Number, default: 1 },
  unit: { type: String, default: '' },
  addedAt: { type: Date, default: Date.now },
});

module.exports = mongoose.model('GroceryItem', GroceryItemSchema);
